diff --git a/src/fake-timers-src.js b/src/fake-timers-src.js
index 2014b2c13a14b15ab324fa1b0f481a5912bc1630..aa2585b94287f008063c3707dfddc22b3fddf65d 100644
--- a/src/fake-timers-src.js
+++ b/src/fake-timers-src.js
@@ -2,14 +2,14 @@
 
 const globalObject = require("@sinonjs/commons").global;
 let timersModule, timersPromisesModule;
-if (typeof require === "function" && typeof module === "object") {
+if (typeof __vitest_required__ !== 'undefined') {
     try {
-        timersModule = require("timers");
+        timersModule = __vitest_required__.timers;
     } catch {
         // ignored
     }
     try {
-        timersPromisesModule = require("timers/promises");
+        timersPromisesModule = __vitest_required__.timersPromises;
     } catch {
         // ignored
     }
@@ -537,7 +537,7 @@ function withGlobal(_global) {
         isPresent.hrtime && typeof _global.process.hrtime.bigint === "function";
     isPresent.nextTick =
         _global.process && typeof _global.process.nextTick === "function";
-    const utilPromisify = _global.process && require("util").promisify;
+    const utilPromisify = _global.process && _global.__vitest_required__ && _global.__vitest_required__.util.promisify;
     isPresent.performance =
         _global.performance && typeof _global.performance.now === "function";
     const hasPerformancePrototype =
