diff --git a/src/fake-timers-src.js b/src/fake-timers-src.js
index 241a7ad3eeb8b5eb691439fab31a87ef13c76345..1891feeece613ca4894ad7cbb29984492d9418af 100644
--- a/src/fake-timers-src.js
+++ b/src/fake-timers-src.js
@@ -2,14 +2,18 @@
 
 const globalObject = require("@sinonjs/commons").global;
 let timersModule, timersPromisesModule;
-if (typeof require === "function" && typeof module === "object") {
+function resolveVitestRequiredTimers() {
+    if (typeof __vitest_required__ === 'undefined') {
+        return;
+    }
     try {
-        timersModule = require("timers");
+        timersModule = timersModule || __vitest_required__.timers;
     } catch {
         // ignored
     }
     try {
-        timersPromisesModule = require("timers/promises");
+        timersPromisesModule =
+            timersPromisesModule || __vitest_required__.timersPromises;
     } catch {
         // ignored
     }
@@ -537,7 +541,7 @@ function withGlobal(_global) {
         isPresent.hrtime && typeof _global.process.hrtime.bigint === "function";
     isPresent.nextTick =
         _global.process && typeof _global.process.nextTick === "function";
-    const utilPromisify = _global.process && require("util").promisify;
+    const utilPromisify = _global.process && _global.__vitest_required__ && _global.__vitest_required__.util.promisify;
     isPresent.performance =
         _global.performance && typeof _global.performance.now === "function";
     const hasPerformancePrototype =
@@ -2704,6 +2708,8 @@ function withGlobal(_global) {
             );
         }
 
+        resolveVitestRequiredTimers();
+
         // eslint-disable-next-line no-param-reassign
         config = typeof config !== "undefined" ? config : {};
         config.shouldAdvanceTime = config.shouldAdvanceTime || false;
