diff --git a/index.d.ts b/index.d.ts
index 986e0d507d88bc033f0a0f0b4dcc40be0635df94..12f2e623eb050ddc55f5562dd9d72af985d497c7 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -2117,7 +2117,7 @@ export const assert: Chai.AssertStatic;
 export const expect: Chai.ExpectStatic;
 
 declare global {
-    interface Object {
-        should: Chai.Assertion;
-    }
+    // interface Object {
+    //     should: Chai.Assertion;
+    // }
 }
