diff --git a/index.d.ts b/index.d.ts
index 5aa018cde4336aca4dadefb8338549c378792e14..1b8136e5fb4c6666a46dbef765c9624d62fdb3a5 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -329,13 +329,15 @@ export interface FakeTimerInstallOpts {
     now?: number | Date | undefined;
 
     /**
-     * An array with names of global methods and APIs to fake. By default, `@sinonjs/fake-timers` does not replace `nextTick()` and `queueMicrotask()`.
-     * For instance, `FakeTimers.install({ toFake: ['setTimeout', 'nextTick'] })` will fake only `setTimeout()` and `nextTick()`
+     * An array with names of global methods and APIs to fake.
+     * For instance, `vi.useFakeTimer({ toFake: ['setTimeout', 'performance'] })` will fake only `setTimeout()` and `performance.now()`
+     * @default everything available globally except `nextTick`
      */
     toFake?: FakeMethod[] | undefined;
 
     /**
-     * The maximum number of timers that will be run when calling runAll() (default: 1000)
+     * The maximum number of timers that will be run when calling runAll()
+     * @default 10000
      */
     loopLimit?: number | undefined;
 
@@ -352,10 +354,16 @@ export interface FakeTimerInstallOpts {
     advanceTimeDelta?: number | undefined;
 
     /**
-     * Tells FakeTimers to clear 'native' (i.e. not fake) timers by delegating to their respective handlers. These are not cleared by
-     * default, leading to potentially unexpected behavior if timers existed prior to installing FakeTimers. (default: false)
+     * Tells FakeTimers to clear 'native' (i.e. not fake) timers by delegating to their respective handlers.
+     * @default true
      */
     shouldClearNativeTimers?: boolean | undefined;
+
+    /**
+     * Don't throw error when asked to fake timers that are not present.
+     * @default false
+     */
+    ignoreMissingTimers?: boolean | undefined;
 }
 
 /**
