diff --git a/dist/rrweb-snapshot.js b/dist/rrweb-snapshot.js
index 3a3b67808308f4b50ad906dccfaaf577b4264742..41d66d31483d9562d4394297b63b36d8e9148843 100644
--- a/dist/rrweb-snapshot.js
+++ b/dist/rrweb-snapshot.js
@@ -1513,6 +1513,12 @@ const pseudoClassPlugin = {
           if (selector.includes(":hover")) {
             rule2.selector += ",\n" + selector.replace(/:hover/g, ".\\:hover");
           }
+          if (selector.includes(":focus-within")) {
+            rule2.selector += ",\n" + selector.replace(/:focus-within/g, ".\\:focus-within");
+          }
+          if (/:focus(?![-\w])/.test(selector)) {
+            rule2.selector += ",\n" + selector.replace(/:focus(?![-\w])/g, ".\\:focus");
+          }
         });
       }
     };
